persistence-query/src/main/scala/org/apache/pekko/persistence/query/scaladsl/CurrentLastKnownSequenceNumberByPersistenceIdQuery.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.pekko.persistence.query.scaladsl
+
+import scala.concurrent.Future
+
+/**
+ * A trait that enables querying the current last known sequence number for a given `persistenceId`.
+ * @since 1.2.0
+ */
+trait CurrentLastKnownSequenceNumberByPersistenceIdQuery extends ReadJournal {
```

Comment:
CurrentLastKnownSequenceNumberQuery ?

Reply:
Just to highlight the intent behind the `Current` in name: It seemed to me that "final" queries like `CurrentEventsByPersistenceIdQuery` are prefixed with 'Current', while continuous corresponding queries like `EventsByPersistenceIdQuery` drop this prefix. The `ByPersistenceId` clarifies the scope of query. Even though a `LastKnownSequenceNumberByPersistenceIdQuery` doesn't exist yet, one could imagine a query that continuously emits the last known sequence number for persistence id in future and thus I had preferred to 1. not block the name already 2. align to existing naming schema.

Reply:
I just think the name is too long, maybe `CurrentLastKnownSequenceNumberQuery`?

Reply:
How about dropping the `Known` from the name - `Current` already implies it in a way. `EventSourcedBehavior.lastSequenceNumber` already exist, so naming would align.

Reply:
Would CurrentLastSequenceNumberQuery work?

Reply:
I think `CurrentLastSequenceNumberByPersistenceIdQuery` is good if we are sticking to the style of having a trait per least amount of methods, which does make sense - for example each journal might want to implement `CurrentLastSequenceNumberByPersistenceIdQuery`, but not necessarily `CurrentLastSequenceNumberBySliceQuery` or `CurrentLastSequenceNumberByTagQuery` if we ever implement them.